app/services/identity_reconciler.go

```diff
@@ -38,12 +38,17 @@ func IdentityReconciler(accountStore data.AccountStore, cfg *app.Config, provide
 			return nil, errors.New("account locked")
 		}
 
+		err = updateUserInfo(accountStore, linkedAccount.ID, providerName, providerUser)
```

Comment:
Should `linkedAccount` have the associated oauth accounts included so it can be passed directly and skip the oauth account lookup in this function?

Reply:
Completely agree with you. In this situation, I think it would be more beneficial to handle the existing technical debt separately. This is because addressing it will require a significant amount of changes everywhere.